### PR TITLE
Link to Flight Job overview for sessions started by Job

### PR DESCRIPTION
**Pull request by VoxSecundus**
This PR adds the ability to navigate to the Flight Job Webapp overview for Desktop sessions which were started by Flight Job. When the session was not launched via Flight Job, the behaviour is as usual.

## Session card

A new row has been added to the session card metadata (see below), showing the job ID as a hyperlink to the Flight Job Webapp page.

Screenshot_2022-06-17_16-02-57

## Session page

The session ID on the session page has been turned into a similarly clickable link (see below), with an icon next to it to indicate that it is a link.

Screenshot_2022-06-17_16-03-15

### CODE REVIEW

**Review by benarmston — CHANGES_REQUESTED** [review]
Thanks for the screenshots.  For the second screenshot, the name would be the link if the session has a name; otherwise it would be the id.  That way the same text "desktop-on-log-4" would be linked in both places.  Obviously, that last point doesn't necessarily hold if the name is changed, but it does by default.

**Comment by VoxSecundus** [comment]
Thanks for the feedback @benarmston; I've updated the page to match what you've described (and `git commit --amend`ed it instead of adding it to a new commit; sorry!)

File: src/SessionPage.js [hunk 1]
```diff
@@ -168,12 +185,17 @@ function Layout({
...
+  const sessionName = <JobLink session={session}>{session?.name}</JobLink>;
+    
   const hostname = session == null ?
     null :
     <span>running on <code className="text-reset">{session.hostname}</code></span>;
   const title = (session == null || session.name == null || session.name === "") ?
-    <span>{sessionId} {hostname}</span> :
-    <span>{session.name} ({sessionId}) {hostname}</span>;
+    <span>{idText} {hostname}</span> :
+    <span>{sessionName} ({idText}) {hostname}</span>;
```
Comment (benarmston) [inline src/SessionPage.js:198]: Looks like its time to extract all of this into a new component.  Perhaps it was time to do so before to be honest.  Could you create a new, say, `HeaderText` component and replace `{title}` on line 210(ish) with a call to `HeaderText`.

File: src/SessionPage.js [hunk 2]
```diff
@@ -154,6 +154,23 @@ function Connected({ id, session }) {
   );
 }
 
+function JobLink({session, children}) {
```
Comment (benarmston) [inline src/SessionPage.js:157]: Good idea extracting this.

File: src/SessionPage.js [hunk 3]
```diff
@@ -154,6 +154,23 @@ function Connected({ id, session }) {
...
+  const jobUrl = `${process.env.REACT_APP_JOBS_CLIENT_BASE_URL}/jobs/${session?.job_id}`;
+  return (
+    <a
+      class="text-light"
+      href={jobUrl}
+      title="Visit Flight Job overview for this session"
+      target="_blank"
+      rel="noreferrer noopener"
```
Comment (benarmston) [inline src/SessionPage.js:165]: None of our other inter-app links open in a new tab.  We should be consistent and not do so with these links either.

File: src/SessionCard.js [hunk 4]
```diff
@@ -27,6 +27,15 @@ function SessionCard({ reload, session }) {
   } else {
     sessionState = session.state;
   }
+  const jobUrl = `${process.env.REACT_APP_JOBS_CLIENT_BASE_URL}/jobs/${session.job_id}`;
+  const jobIdEntry = session.job_id ?
+    <MetadataEntry
+      name="Job ID"
+      value={session.job_id}
+      valueTitle="Visit Flight Job overview for this session"
+      format={id => <a href={jobUrl} target="_blank" rel="noreferrer noopener">{id}</a> }
```
Comment (benarmston) [inline src/SessionCard.js:36]: See other comment about opening in new tabs.

**Review by benarmston — CHANGES_REQUESTED** [review]
A few comments for you to look at please.

File: src/HeaderText.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    session?.name;
+  
+  const hostname = session == null ?
+    null :
+    <span> running on <code className="text-reset">{session.hostname}</code></span>;
+  const title = ( session == null || session.name == null || session.name === "") ?
+    <span>{idText} {hostname}</span> :
+    <span>{sessionName} ({idText}) {hostname}</span>;
+
+  return (title)
```
Comment (benarmston) [inline src/HeaderText.js:35]: Now that this is extracted to its own component we have a chance to refactor to improve the clarity of what cases it is handling and how.

The cases that its handling are:

1. The session is null (presumably its being loaded).
2. The session has a name and a job id.
3. The session has a name, but not a job id.
4. The session does not have a name, but does have a job id.
5. The session does not have a name and does not have a job id.

Early returns and possibly extracting additional components might help here.

**Review by benarmston — CHANGES_REQUESTED** [review]
Can you have a go at cleaning up the new `HeaderText` component.  It should be simpler to do so now that its its own component.

**Review by benarmston — CHANGES_REQUESTED** [review]
Looks like you're on the right lines...

**Review by benarmston — APPROVED** [review]
I've added a commit which renames `HeaderText` to `SessionHeaderText` and adjusts the tooltips and icons.  Nothing major.

This looks good to me know.

**Pull request merged**